OpenRA.Mods.Common/Widgets/Logic/Ingame/IngameMenuLogic.cs

```diff
@@ -88,14 +88,38 @@ public IngameMenuLogic(Widget widget, World world, Action onExit, WorldRenderer
 
 			abortMissionButton.OnClick = () =>
 			{
-				if (world.IsGameOver)
+				hideMenu = true;
+				Action restartAction = null;
+				if (world.IsReplay || world.Map.Visibility == MapVisibility.MissionSelector)
 				{
-					onQuit();
-					return;
+					var iop = world.WorldActor.TraitsImplementing<IObjectivesPanel>().FirstOrDefault();
+					var exitDelay = iop != null ? iop.ExitDelay : 0;
+					restartAction = () =>
+					{
+						Ui.CloseWindow();
+						if (mpe != null)
+						{
+							if (Game.IsCurrentWorld(world))
+								mpe.Fade(MenuPaletteEffect.EffectType.Black);
+							exitDelay += 40 * mpe.Info.FadeLength;
+						}
+
+						Game.RunAfterDelay(exitDelay, () =>
+						{
+							Game.RestartGame();
+						});
+					};
 				}
 
-				hideMenu = true;
-				ConfirmationDialogs.PromptConfirmAction("Abort Mission", "Leave this game and return to the menu?", onQuit, showMenu);
+				ConfirmationDialogs.PromptConfirmAction(
+					title: "Abort Mission",
+					text: "Leave this game and return to the menu?",
+					onConfirm: onQuit,
+					onCancel: showMenu,
+					confirmText: "Abort",
```

Comment:
I was just copying over what was mostly there but I do see the value in showing a restart button even if you won the game. What if you wanted to restart the mission to complete a secondary objective you missed? So what would you rather happen since it appears the old behavior was after you win you have the option of staying "cancel" or leaving "abort".